### PR TITLE
Removing "DeprecationWarning: Initializing a coupling object through a couplingdict is deprecated"

### PR DESCRIPTION
**Pull request by 1ucian0**
Many of the tests I wrote are using the old format for creating a coupling map. This PR moves them to the new format. Also, if the first positional argument of `Coupling()` is a list, use it as a list.

### CODE REVIEW

File: test/python/transpiler/test_basic_mapper.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def test_trivial_case(self):
 
          Coupling map: [1]--[0]--[2]
         """
-        coupling = Coupling({0: [1, 2]})
+        coupling = Coupling([(0, 1), (0, 2)])
```
Comment (ajavadia) [inline test/python/transpiler/test_basic_mapper.py]: It seems the format specified in Coupling is list of lists, not list of pairs. 
Maybe a reason for this is JSON encoding. I'm not fully sure.
But until we discuss and potentially change that in all Coupling constructors, can you make your code here consistent with the Coupling definition?

**Comment by jaygambetta** [comment]
Yeah, i agree. It should be of the form 

```
coupling_map = [[1, 0], [2, 0], [2, 1], [3, 2], [3, 4], [4, 2]]
```

Also, i think we should call this `CouplingMap` and `coupling_map` as this is what we use in the schema

**Comment by 1ucian0** [comment]
I can do the rename in this very same PR. About `[(1, 0), (2, 0)]` vs 
`[[1, 0] , [2, 0]]` are semantically equivalent for our porpoises and I think the first form is easier to read.

**Comment by jaygambetta** [comment]
i agree they are the same for this purpose but it is not what is returned from the backend. configruation or schema. I like having the consistency but i let @ajavadia weigh in.

**Comment by 1ucian0** [comment]
the `[(1, 0), (2, 0)]` vs `[[1, 0] , [2, 0]]` fixed in effdf7f

**Pull request merged**